fedbiomed/node/round.py

```diff
@@ -114,21 +114,55 @@ def __init__(
         self._optim_aux_var = {}  # type: Dict[str, Dict[str, Any]]
         self._node_state_manager = NodeStateManager(environ['DB_PATH'])
 
-    def initialize_validate_training_arguments(self) -> None:
+    def _initialize_validate_training_arguments(self) -> None:
         """Validates and separates training argument for experiment round"""
 
         self.training_arguments = TrainingArgs(self.training_kwargs, only_required=False)
         self.testing_arguments = self.training_arguments.testing_arguments()
         self.loader_arguments = self.training_arguments.loader_arguments()
 
     def initialize_node_state_manager(self, previous_state_id: Optional[str] = None):
-        """Initializes [`NodeStateManager`][fedbiomed.node.node_state_manager.NodeStateManager]. 
+        """Initializes [`NodeStateManager`][fedbiomed.node.node_state_manager.NodeStateManager].
 
         Args:
             previous_state_id (optional): previous state_id from which to load `Node` state.
-            Defaults to None (which is the value for `Round` 0).
+                Defaults to None (which is the value for `Round` 0).
         """
-        self._node_state_manager.initialize(previous_state_id=previous_state_id)
+
+        self._node_state_manager.initialize(previous_state_id=previous_state_id,
+                                            testing=not self.training)
+
+    def initialize_validate_training_arguments(self) -> Dict[str, Any]:
+        """Initialize and validate requested experiment/training arguments.
+
+        Returns:
+            Returns the corresponding node message, training reply instance
+        """
+        try:
+            self._initialize_validate_training_arguments()
+        except FedbiomedUserInputError as e:
+            return self._send_round_reply(success=False, message=repr(e))
+        except Exception as e:
+            msg = 'Unexpected error while validating training argument'
+            logger.debug(f"{msg}: {repr(e)}")
+            return self._send_round_reply(success=False, message=f'{msg}. Please contact system provider')
+
+    def initialize_arguments(self,
```

Comment:
Thank you @ybouilla for the nice fix !

As discussed, we may simplify a little bit the code by removing some very short functions:
- `initialize_node_state_manager` (merge code in `initialize_arguments`)
-  `_initialize_validate_training_arguments` (merge code in `initialize_validate_training_arguments`)

You may keep `initialize_validate_training_arguments` (and maybe make it private ?).

Reply:
Regarding the points you spotted above, I 'd like to keep `initialize_validate_training_arguments` and make it private.
Concerning the `_save_round_state` method place, i undo the changes. Hence, if `test_ratio = 1`, Node state will still be saved even if no training is performed. We made this decision for simplicity sake.

Reply:
However, TrainRequest fails if `test_ratio=1` and `test_on_local_updates=True`. This bug is not that simple to fix and a issue has be opened #901